de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/invoice/impl/InvoiceService.java

```diff
@@ -67,12 +69,25 @@ public boolean hasArchive(@NonNull final InvoiceId invoiceId)
 			return Optional.empty();
 		}
 
-		List<I_AD_Archive> lastArchive = archiveDAO.retrieveLastArchives(Env.getCtx(), TableRecordReference.of(invoiceRecord), 1);
+		final List<I_AD_Archive> lastArchive = archiveDAO.retrieveLastArchives(Env.getCtx(), TableRecordReference.of(invoiceRecord), 1);
 		if (Check.isEmpty(lastArchive))
 		{
 			return Optional.empty();
 		}
 		return Optional.of(lastArchive.get(0));
 	}
 
+	public ResponseEntity.HeadersBuilder<?> reverseInvoice(@NonNull final InvoiceId invoiceId)
```

Comment:
what is the  `.HeadersBuilder<?> about?
i never saw it in a return type.
why do we need it?

Reply:
It's because in `de.metas.rest_api.invoicecandidates.impl.InvoicesRestControllerImpl#revertInvoice` we do `invoiceService.reverseInvoice(invoiceId).build();`

Should i build the answer inside `InvoiceService`?

Reply:
yes, build is inside `InvoiceService`, unless there is something you want to do with it later, which is currently not the case